### PR TITLE
Improve awc websocket docs

### PR DESCRIPTION
**Pull request by popzxc**
## PR Type

Docs

## PR Checklist

Check your PR fulfills the following:

- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [ ] A changelog entry has been made for the appropriate packages.
- [x] Format code with the latest stable rustfmt

## Overview

When I first tried to use `awc`, I had some problems with understanding how to use it with WebSocket.

I thought an example will be very helpful for readers.

### CODE REVIEW

**Comment by popzxc** [comment]
Hm. I'm not sure why CI fails on windows, but passes on Linux / macOS. It doesn't seem that doctest has some platform-specific code.

**Comment by Neopallium** [comment]
Looks like an issue getting a cert for TLS:

"the handshake failed: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:ssl\\statem\\statem_clnt.c:1924:: unable to get local issuer certificate"

I don't think it is related to this PR.

**Comment by Neopallium** [comment]
Would it work with "ws://echo.websocket.org" ws instead of wss?

File: awc/src/lib.rs [hunk 1]
```diff
@@ -193,7 +193,8 @@ impl Client {
         self.request(Method::OPTIONS, url)
     }
 
-    /// Construct WebSockets request.
+    /// Initialize a WebSockets connection.
```
Comment (robjtede) [inline awc/src/lib.rs]:
```suggestion
    /// Initialize a WebSocket connection.
```

File: awc/src/lib.rs [hunk 2]
```diff
@@ -193,7 +193,8 @@ impl Client {
...
+    /// This method returns a builder structure to establish a WebSocket connection.
```
Comment (robjtede) [inline awc/src/lib.rs]:
```suggestion
    /// Returns a WebSocket connection builder.
```

File: awc/src/ws.rs [hunk 3]
```diff
@@ -1,4 +1,33 @@
 //! Websockets client
+//!
+//! This module contains type definitions required to use [`awc::Client`](../struct.Client.html) as a WebSocket client.
```
Comment (robjtede) [inline awc/src/ws.rs]:
```suggestion
//! Type definitions required to use [`awc::Client`](../struct.Client.html) as a WebSocket client.
```

File: awc/src/ws.rs [hunk 4]
```diff
@@ -1,4 +1,33 @@
...
+//! To use `awc::Client`
```
Comment (robjtede) [inline awc/src/ws.rs]:
```suggestion
```

**Review by Neopallium — APPROVED** [review]
Looks good.

File: awc/src/ws.rs [hunk 5]
```diff
@@ -1,4 +1,33 @@
...
+//!     connection
+//!         .send(ws::Message::Text("Echo".to_string()))
+//!         .await
+//!         .unwrap();
+//!     let response = connection.next().await.unwrap().unwrap();
+//!
+//!     assert_eq!(response, ws::Frame::Text("Echo".as_bytes().into()));
+//! }
+//! ```
+//!
```
Comment (robjtede) [inline awc/src/ws.rs]:
```suggestion
```

**Review by robjtede — CHANGES_REQUESTED** [review]
just some wording adjustments, otherwise LGTM

**Review by robjtede — APPROVED** [review]
thanks for the contribution 👍

**Pull request merged**